statistics/importResults.js

```diff
@@ -0,0 +1,178 @@
+const mongoose = require('mongoose');
+const xlsx = require('node-xlsx');
+const Result = require('./models/Result');
+const Course = require('./models/Course');
+
+
+const courseProperties = [
+  'courseCode',
+  'courseName',
+  'programShort',
+  'programLong',
+];
+const filename = 'Tentastatistik.xlsx';
+
+//Main Async function
+async function main() {
+  mongoose.set('useCreateIndex', true);
+  mongoose.set('useNewUrlParser', true);
+  console.log("Connecting to the database...");
+  await mongoose.connect('mongodb://localhost/stats', {
+    useNewUrlParser: true,
+    useUnifiedTopology: true
+  });
+  console.log("Connected to database!");
+
+  let courses = exportDataFromSpreadsheet();
+  let results = [];
+
+  //Loop through all the courses in the datasheet one by one.
+  for(key in courses) {
+    let co = courses[key];
+    let resultsInDb = [];
+    let updatedResults = [];
+    //Try to find that course in the database by its course code.
+    let dbCourse = await Course.findOne({courseCode: co.courseCode}, {results: 1});
+
+    if(dbCourse != null) {
+      for(let i in dbCourse.results) {
+        //Check what exams are already in the databse (no need to add them again).
```

Comment:
Does this also work if exam results are spread out through different files?
Let's say that A.xlsx contains half the results of an exam that ran 2021-11-27 and later file B.xlsx comes in that has the other half of that same exam.
Will that half be ignored when B.xlsx is loaded?

Reply:
Nice catch, it would indeed be the case. It's a consequence of an assumption that all exams will be grouped together in the same document and not in multiple ones. This will almost always be the case but it can be fixed so I will look into it.